pkg/ccl/changefeedccl/alter_changefeed_stmt.go

```diff
@@ -377,15 +385,51 @@ func generateNewTargets(
 				return nil, nil, hlc.Timestamp{}, nil, err
 			}
 
-			_, withInitialScan := targetOpts[changefeedbase.OptInitialScan]
-			_, noInitialScan := targetOpts[changefeedbase.OptNoInitialScan]
-			if withInitialScan && noInitialScan {
+			var withInitialScan bool
+			initialScanType, initialScanSet := targetOpts[changefeedbase.OptInitialScan]
+			_, noInitialScanSet := targetOpts[changefeedbase.OptNoInitialScan]
+			_, initialScanOnlySet := targetOpts[changefeedbase.OptInitialScanOnly]
+
+			if initialScanSet {
+				if initialScanType == `no` || (initialScanType == `only` && !originalInitialScanOnlyOption) {
+					withInitialScan = false
+				} else {
+					withInitialScan = true
+				}
+			} else {
+				withInitialScan = false
+			}
+
+			if initialScanType != `` && initialScanType != `yes` && initialScanType != `no` && initialScanType != `only` {
+				return nil, nil, hlc.Timestamp{}, nil, pgerror.Newf(
+					pgcode.InvalidParameterValue,
+					`cannot set initial_scan to %q`, changefeedbase.OptInitialScan,
+				)
+			}
+
+			if initialScanSet && noInitialScanSet {
```

Comment:
Style nit perhaps, seems like “no two options specified” logic could be something tighter, in the spirit of an xor?

Reply:
Sounds good. Small nit, like I said -- if the alternative would be longer or harder to grok, that’s worse! Thanks.